### PR TITLE
FS-3685-view-assessment-activity-history-for-an-application

### PR DESCRIPTION
**Pull request by ramsharma-prog**
As an assessor (QA User) I need to see the application activity log so that I can check the quality of my work

- [X] Unit tests and other appropriate tests added or updated
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Unit tests added 

Screenshot 2023-11-24 at 19 46 26

Screenshot 2023-11-24 at 19 43 33

### CODE REVIEW

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [inline app/blueprints/assessments/activity_trail.py:44]: We prob don't need to log that we're formatting the date, could remove that

Also, you could supress the error since the end return the date string at the end anyways:

```suggestion
    @staticmethod
    def _format_date(date_str):
        if date_str:
            with contextlib.suppress(ValueError)
                return datetime.fromisoformat(date_str).strftime(
                    "%Y-%m-%d %H:%M:%S"
                )
        return date_str
```

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [inline app/blueprints/assessments/activity_trail.py:53]: Do we need `date_created`? Can't we just use the `created_at` from the parent class?

https://github.com/communitiesuk/funding-service-design-assessment/blob/52e8a270289a80b43883f766c1220ed253372d28/app/blueprints/tagging/models/tag.py#L51

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [inline app/blueprints/assessments/activity_trail.py:100]: Probably dont need this log

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [inline app/blueprints/assessments/activity_trail.py:161]: Another log

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [inline app/blueprints/assessments/activity_trail.py:198]: You could use a set comprehension for this, since it doesn't contain duplicates

```suggestion
    user_ids = {item.user_id for item in list_data if item.user_id}
    # user_list = list(user_ids)  # if you need it as a list afterwards
```

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [inline app/blueprints/assessments/activity_trail.py:189]: Could we type or re-name list data? to see what it represents

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [inline app/blueprints/assessments/activity_trail.py:214]: `account_info` looks to be immediately re-assigned - do we need it here?

File: app/blueprints/assessments/routes.py
Comment (tferns) [inline app/blueprints/assessments/routes.py:760]: Can we use the same convention rather than `all_...` & `_...`?

File: app/blueprints/assessments/activity_trail.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/activity_trail.py:44]: Thank you. Sure!

File: app/blueprints/assessments/activity_trail.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/activity_trail.py:100]: Thank you. Sure!

File: app/blueprints/assessments/activity_trail.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/activity_trail.py:198]: Aah yes! Thank you!

File: app/blueprints/assessments/activity_trail.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/activity_trail.py:233]: Yes, sorry forgot to update

File: app/blueprints/assessments/routes.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/routes.py:750]: Will remove these

File: app/blueprints/assessments/routes.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/routes.py:760]: Yes! will do 😄 . Didnt realise

File: app/blueprints/assessments/templates/activity_trail.html
Comment (ramsharma-prog) [reply app/blueprints/assessments/templates/activity_trail.html:20]: Yes we need the header to display  the banner summary and other information unless I am missing something? 🤔

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [reply app/blueprints/assessments/activity_trail.py:49]: But couldn't we make that function use created_at instead? could make it a property method too if all it needs to do is access `date_created`

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [reply app/blueprints/assessments/activity_trail.py:197]: yea but in the function, it has `account_info: list = None` but then the line after `account_info = get_user_info(list_data, state)`, so the first part is unnecessary, cause it's immediately re-assigned

File: app/blueprints/assessments/templates/activity_trail.html
Comment (tferns) [reply app/blueprints/assessments/templates/activity_trail.html:20]: Yea, but this looks copy pasted, rather than using the `header.html` like this: https://github.com/communitiesuk/funding-service-design-assessment/blob/e631c389272ded1e20889f84b07b68137b944602/app/blueprints/tagging/templates/change_tags.html#L4-L6

File: app/blueprints/assessments/activity_trail.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/activity_trail.py:197]: Aah, I see what you mean 😆 .  Sorry missed that part 🙏

File: app/blueprints/assessments/activity_trail.py
Comment (ramsharma-prog) [reply app/blueprints/assessments/activity_trail.py:49]: Yes, that is one way of doing it. I just wanted to handle it within the class and thought it looks cleaner 🤔

File: app/blueprints/assessments/templates/activity_trail.html
Comment (tferns) [reply app/blueprints/assessments/templates/activity_trail.html:20]: Ah! That’s exactly the answer I was looking for, nice 👍

File: app/blueprints/assessments/activity_trail.py
Comment (tferns) [reply app/blueprints/assessments/activity_trail.py:49]: Ah cool can leave if you think this way is better

**Review by tferns — DISMISSED** [review]
Looks good rbut in your custom header, I’d watch out calling banner summary, changes to that macro were merged to main - so I’d pull first and fix them before merging

**Comment by ramsharma-prog** [comment]
> Looks good rbut in your custom header, I’d watch out calling banner summary, changes to that macro were merged to main - so I’d pull first and fix them before merging

Yes I pulled the changes. Thanks for letting me know 😃

**Review by rayhan15243 — APPROVED** [review]
Approved, but only merge once @tferns gives a green tick too.

**Pull request merged**
